worker/uniter/resolver.go

```diff
@@ -52,6 +52,13 @@ func (s *uniterResolver) NextOp(
 		return nil, resolver.ErrTerminate
 	}
 
+	// If the unit has completed a pre-series-upgrade hook (as noted
+	// by its state) then the uniter should idle in the face of
+	// all remote state changes - the unit is waiting to be shutdown.
+	if localState.UpgradeSeriesStatus == model.UnitCompleted && remoteState.UpgradeSeriesStatus == model.UnitCompleted {
```

Comment:
When the resolver loop hits this state, it should only return an `ErrNoOperation`. This should cause the Uniter to idle until the next state change. At this point, the uniter is essentially waiting to be shut down. Which is why the only state change it accepts is that which results in `ErrTerminate`.